tests/integration/conftest.py

```diff
@@ -121,3 +126,50 @@ def jenkins_test_job_xml() -> str:
         </project>
         """
     )
+
+
+@pytest_asyncio.fixture(scope="module", name="machine_controller")
+async def machine_controller_fixture() -> Controller:
+    """The lxd controller."""
+    controller = Controller()
+    await controller.connect_controller("localhost")
+
+    return controller
+
+
+@pytest.fixture(scope="module", name="machine_model_name")
+def machine_model_name_fixture(request: pytest.FixtureRequest) -> str:
+    """The name for machine model."""
+    # This is taken from the same logic in pytest_operator plugin.py _generate_model_name
+    # to match the ops test naming.
+    module_name = request.module.__name__.rpartition(".")[-1]
+    suffix = "".join(choices(ascii_lowercase + digits, k=4))  # nosec
```

Comment:
What about using `secrets.token_hex(2)`?

Reply:
that sounds better! thank you!